### PR TITLE
Refactor `_add_grub_bootloader()` installation

### PR DESCRIPTION
**Pull request by codefiles**

### CODE REVIEW

**Review by Torxed — APPROVED** [review]

**Comment by codefiles** [comment]
I guess I'll wait until #2015 is merged to resolve conflicts.

**Comment by svartkanin** [comment]
@codefiles apologies for the delay, trying to catchup with PRs atm, as there are multiple ones getting merged now there are conflicts unfortunately. Would you mind resolving them and then we can merge these as well

**Review by Torxed — APPROVED** [review]

**Pull request merged**
